extensions/vscode/src/views/homeView.ts

```diff
@@ -1189,6 +1209,174 @@ export class HomeViewProvider implements WebviewViewProvider, Disposable {
+    // get the configs which are filtered for the entrypoint in that projectDir
+    // determine a list of deployments that use those filtered configs
+
+    let contentRecordList: (ContentRecord | PreContentRecord)[] = [];
+    const getContentRecords = new Promise<void>(async (resolve, reject) => {
+      try {
+        const response = await api.contentRecords.getAll({
+          dir,
+          recursive: false,
+        });
+        response.data.forEach((c) => {
+          if (!isContentRecordError(c)) {
+            contentRecordList.push(c);
+          }
+        });
+      } catch (error: unknown) {
+        const summary = getSummaryStringFromError(
+          "handleFileInitiatedDeployment, contentRecords.getAll",
+          error,
+        );
+        window.showInformationMessage(
+          `Unable to continue due to deployment error. ${summary}`,
+        );
+        return reject();
+      }
+      return resolve();
+    });
+
+    let configMap = new Map<string, Configuration>();
+    const getConfigurations = new Promise<void>(async (resolve, reject) => {
+      try {
+        const response = await api.configurations.getAll({
+          dir,
+          entrypoint,
+          recursive: false,
+        });
+        let rawConfigs = response.data;
+        rawConfigs.forEach((c) => {
+          if (!isConfigurationError(c)) {
+            configMap.set(c.configurationName, c);
+          }
+        });
+      } catch (error: unknown) {
+        const summary = getSummaryStringFromError(
+          "handleFileInitiatedDeployment, configurations.getAll",
+          error,
+        );
+        window.showInformationMessage(
+          `Unable to continue with API Error: ${summary}`,
+        );
+        return reject();
+      }
+      resolve();
+    });
+
+    const apisComplete = Promise.all([
+      // getConfigurationInspections,
+      getContentRecords,
+      getConfigurations,
+    ]);
+
+    showProgress(
+      "Initializing::handleFileInitiatedDeployment",
+      Views.HomeView,
+      apisComplete,
+    );
+
+    try {
+      await apisComplete;
+    } catch {
+      // errors have already been displayed by the underlying promises..
+      return undefined;
+    }
+
+    // Build up a list of compatible content records with this entrypoint
+    const compatibleContentRecords: (ContentRecord | PreContentRecord)[] = [];
+    contentRecordList.forEach((c) => {
+      if (configMap.get(c.configurationName)) {
+        compatibleContentRecords.push(c);
+      }
+    });
+
+    // console.log(
+    //   `compatibleContentRecords: # found: ${compatibleContentRecords.length}`,
+    // );
+
+    // if no deployments, create one
+    if (!compatibleContentRecords.length) {
+      // call new deployment
+      const selected = await this.showNewDeploymentMultiStep(
+        Views.HomeView,
+        dir,
+        entrypoint,
+      );
+      if (selected) {
+        window.showInformationMessage(
```

Comment:
Noted this in conversations, but once we get these deploying on-click I think we can remove these window notifications.

Reply:
yes, they will be.